src/components/form/FToggle.vue

```diff
@@ -0,0 +1,348 @@
+  
+.FToggle--disabled
+  cursor default
+  .FToggle__track
+    outline-width rem(1) !important
+    outline-color var(--FToggle--border-color) !important
+
+  .FToggle__label,
+  .FToggle__labelText
+    cursor default
+    color var(--color--neutral--dark-1) !important
+  
+.FToggle--error
+    color var(--FToggle--error-color)
+</style>
+  
+<script setup lang="ts">
+import { computed, ref, watch } from 'vue';
+import FIcon from '@/components/FIcon.vue';
+import FField from '@/components/form/FField.vue';
+import { getCssColor } from '@/utils/getCssColor';
+import { useFieldWithValidation } from '@/composables/useFieldWithValidation';
+import { useInputEventBindings } from '@/composables/useInputEventBindings';
+
+export interface FToggleProps {
+  /**
+   * Checkbox v-model value
+   * @model
+   */
+  modelValue?: boolean;
+  /**
+   * Label of the toggle
+   */
+  label?: string;
+  /**
+   * Position the label to the right of the toggle
+   */
+  reverse?: boolean;
+  /**
+   * Disable the toggle usage
+   */
+  disabled?: boolean;
+  /**
+   * Hide or not the hint / error message
+   */
+  hideHint?: boolean;
+  /**
+   * A hint to display under the toggle
+   */
+  hint?: string;
+  /**
+   * Icon, displayed before the hint
+   */
+  hintIcon?: Icon | null;
+  /**
+   * Text color of the hint
+   */
+  hintTextColor?: Color;
+  /**
+   * Background color of the toggle
+   */
+  color?: Color;
+  /**
+   * Background color of the thumb
+   */
+  thumbColor?: Color;
+  /**
+   * Text color of the toggle
+   */
+  textColor?: Color;
+  /**
+   * Color of the border
+   */
+  borderColor?: Color;
+  /**
+   * Color of the border on hover
+   */
+  hoverBorderColor?: Color;
+  /**
+   * Background color when toggle is toggled
+   */
+  toggledColor?: Color;
+  /**
+   * Border color when toggle is toggled
+   */
+  toggledBorderColor?: Color;
+  /**
+   * Don't show a checkmark icon when toggle is on
+   */
+  noCheckmarkIcon?: boolean;
+  /**
+   * Smaller label text
+   */
+  smallText?: boolean;
+  /**
+   * Hint error color
+   */
+  errorColor?: Color;
+  /**
+   * Field name. Used in a form context
+   */
+  name?: string;
+  /**
+   * Event that triggers validation
+   */
+  validationTrigger?: 'change' | 'focus' | 'blur';
+  /**
+   * Rules form validation
+   */
+  rules?: ValidationRule | ValidationRule[];
+  /**
+   * Message to use as hint when validation fails
+   */
+  errorMessage?: string;
+  /**
+   * Whether the input should be validated on mount
+   */
+  validateOnMount?: boolean;
+}
+
+const props = withDefaults(defineProps<FToggleProps>(), {
+  modelValue: false,
+  borderColor: 'neutral--light-2',
+  toggledBorderColor: 'secondary',
+  toggledColor: 'secondary',
+  color: 'neutral--light-2',
+  thumbColor: 'neutral--light-5',
+  disabled: false,
+  errorColor: 'danger',
+  errorMessage: '',
+  hideHint: false,
+  noCheckmarkIcon: false,
+  smallText: false,
+  hint: '',
+  hintTextColor: 'neutral--dark-4',
```

Comment:
```suggestion
  hintTextColor: 'neutral--dark-3',
```

Reply:
We used 'neutral--dark-3' color on other Form elements, I think it would be better to keep it consistent ;)

Reply:
We actually use `neutral--dark-4` for the `textColor` and `hintTextColor` for all components with an exception for `FCheckbox` and `FRadio`, ideally those 2 should have `neutral--dark-4` aswell, but I think we should keep them like this for now because it would break the apps using them with the default value :sweat_smile: